lib/HLSL/DxilNoOptLegalize.cpp

```diff
@@ -120,10 +122,26 @@ class DxilNoOptSimplifyInstructions : public ModulePass {
               I->eraseFromParent();
               Changed = true;
             }
+          } else if (PHINode* Phi = dyn_cast<PHINode>(I)) {
+            if (Phi->getNumIncomingValues() == 1) {
```

Comment:
We should probably just call `llvm::SimplifyInst` on the Phi. That will handle both the one incoming value case and the multiple incoming values when they are all the same.

Reply:
Done.